### PR TITLE
eCommerce Site

### PR DESCRIPTION
**Pull request by Yatin-Bajaj**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+    <div>
+      <Header />
+      <div className={classes.main}>
+        <Routes>
+          <Route path="/" element={<Navigate replace to={"/home"} />} />
+          <Route path="/home" element={<Home />} />
+          <Route path="/book/:id" element={<BookDetails />} />
+          <Route path="/cart" element={<Cart />} />
+          <Route path="/orders" element={<Order />} />
+          <Route path="*" element={<div>404 PAGE NOT FOUND</div>} />
```
Comment (sahil1504001) [inline src/App.tsx:20]: moved 404 msg to const

File: src/components/UI/Button.module.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.btn {
+    color: #f7f7f7;
+    background-color: rgb(26, 156, 176);
```
Comment (sahil1504001) [inline src/components/UI/Button.module.css:3]: try to declare and use it from variables if its us-used

File: src/components/books/BookDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                    <div className={`${classes.space}`}>Page Count: {pageCount}</div>
+                    <div className={`${classes.space}`}>Categories: {categories}</div>
+                    <div className={`${classes.space} ${classes.cart_btn}`}>
+                        <Button onClick={cartHandler}>Add to cart</Button>
+                        <Button onClick={buyNowHandler}>Buy Now</Button>
+                    </div>
+                    <div className={classes["book_detail_desc"]}>
+                        Description {description}
+                    </div>
+                </div>
```
Comment (sahil1504001) [inline src/components/books/BookDetails.tsx:65]: keep strings in const

File: src/components/cart/PaymentInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            </div>
+            <div>
+                <p>Total</p>
+                <p>{total}</p>
+            </div>
+
+            <div>
+                <Button onClick={checkoutHandler}>Checkout ✔</Button>
+            </div>
+        </div>
```
Comment (sahil1504001) [inline src/components/cart/PaymentInfo.tsx:59]: keep strings in const

File: src/components/cart/ShippingAddressForm.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                paylod: event.target.value,
+            });
+        };
+    };
+    
+    const onSubmitHandler: React.FormEventHandler<HTMLFormElement> = (
+        event: React.FormEvent<HTMLFormElement>
+    ) => {
+        event.preventDefault();
+        localStorage.setItem("shippingAddress", JSON.stringify(state));
```
Comment (sahil1504001) [inline src/components/cart/ShippingAddressForm.tsx:46]: keep key in const

File: src/components/cart/ShippingAddressForm.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                        name={inputElement.name}
+                        label={inputElement.label}
+                        value={`${state[inputElement.valueType]}`}
+                        readOnly={!isFormEditable ? true : false}
+                        onChange={onChangeHandler(inputElement.ActionType)}
+                    />
+                ))}
+
+                <Button className={classes.control} onClick={onEditHandler}>
+                    {isFormEditable ? "Save Address" : "Edit Address"}
```
Comment (sahil1504001) [inline src/components/cart/ShippingAddressForm.tsx:64]: move button names to const

File: src/components/cart/ShoppingBag.module.css [hunk 7]
```diff
@@ -0,0 +1,49 @@
+.checkoutDetailsContainer {
```
Comment (sahil1504001) [inline src/components/cart/ShoppingBag.module.css:1]: for next project, can u pls try to use? vh and vw?

File: src/store/api.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+export const getOrders = (): Promise<OrderType[]> => {
+    return fetch(`http://localhost:3500/orders`)
+        .then((response) => response.json())
+        .then((data) => data);
+};
+
+export const createOrder = (order: OrderType) => {
+    return fetch(`http://localhost:3500/orders`, {
```
Comment (sahil1504001) [inline src/store/api.ts:46]:
```
API_BASE_HREF: 'http://localhost:3500/',
ROUTES: {
  ORDERS: '/order'
} 
```

use like this

File: src/store/api.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                pageCount: book.volumeInfo.pageCount,
+                categories: book.volumeInfo.categories.join(","),
+                description: book.volumeInfo.description,
+                imageUrl: book.volumeInfo.imageLinks.smallThumbnail,
+            };
+        });
+};
+
+export const getOrders = (): Promise<OrderType[]> => {
+    return fetch(`http://localhost:3500/orders`)
```
Comment (sahil1504001) [inline src/store/api.ts:40]: for angular, use HttpClient from '@angular/common/http' to know advantages and disadvantages of it

**Comment by Yatin-Bajaj** [comment]
All the requirements have been fulfilled and all the changes made look's good to me and mentor Sahil both.

**Pull request merged**